### DISCUSSION TITLE
Recursive State

### DISCUSSION THREAD

**Discussion by mstfy** [pointfreeco/swift-composable-architecture#474] [Q&A]
It is fairly common to have one view pushes same view to navigation stack. For example A ProductDetailView can contain similar products section. And tapping a product in similar products sections pushes a new ProductDetailView to navigation view. It is easy to implement it in vanillia SwiftUI. Just create a new instance and give it to destination of NavigationLink.

But I am having struggle to achieve same thing in TCA. I need to embed the child view's state to parent view's state in order to trigger navigation. This is the ProductDetailState:

```
struct ProductDetailState: Equatable {
    var isNavigationActive = false
    var similarProducts: [Product] = []
    var similarProductState: ProductDetailState?
}
```

But because it is struct and structs can not recursively contain their instances I get compile error. I tried to convert state to final class but this time views didn't update.

Am I missing something? How can achieve this in TCA?

**Comment by RabugenTom** [comment]
[There is an example about recursion in the case studies](https://github.com/pointfreeco/swift-composable-architecture/blob/main/Examples/CaseStudies/SwiftUICaseStudies/04-HigherOrderReducers-Recursion.swift). Maybe it'll help.

**Comment by ChaseStas** [comment]
[Here is an example with two states](https://github.com/ChaseStas/recurse-navigation-tco)

**Comment by stephencelis** [comment]
Going to convert this to a discussion, but the examples above should help 🙂
